### ISSUE TITLE
Add SPDX output support

### ISSUE THREAD

**Issue by ezamriy** [AlmaLinux/alma-sbom#11]
[SPDX](https://spdx.dev/) is the only one ISO certified SBOM metadata format and an industry standard along with CycloneDX, so we need to support it in the AlmaLinux SBOM tooling.

**Comment by javihernandez** [comment]
Breaking this into smaller pieces. We need to:
* First, we need to identify what information we shall use when creating spdx-formatted sboms. We'd need to understand the SPDX format, what can we include, what's not, etc - Ideally we could create a draft proposal that we can review and agree on
* Find discrepancies between our current CycloneDX and potential improvements that need to be done in alma-sbom
* Extend alma-sbom in a way that we add support new output format (SPDX) into the different file format types that spdx-tools support

Feel free to ask any questions if needed ;)

**Comment by m10k** [comment]
I finally managed to add an SPDX module to libsbom and test it in my environment. On the surface, the generated SPDX files look fine, but there are still a number of things that need to be addressed:

1. Under what license should SBOM files be released?
2. We need to determine a namespace for SPDX files (i.e. a location where SPDX files are stored, like "https://security.almalinux.org/spdx")
3. The module depends on the latest spdx-tools from Github because the release on PyPi is outdated
4. Packages in SPDX SBOMs must contain the download location, which should be the URL of the package on a package mirror. I need to implement a function that converts a purl (something like "pkg:rpm/almalinux/bash@5.1.8") to something that the SPDX Validator accepts
5. I have written the spdx module so it generates annotations for all properties like "almalinux:sbom:casHash". I need to double-check if this is correct.

Since there are still issues with the code, I've created a draft pull request for now:
https://github.com/AlmaLinux/alma-sbom/pull/16
